src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -443,6 +444,20 @@ object PruneDeadFields {
         val dep = requestedType.copy(rvRowType = unify(child.typ.rvRowType,
           requestedType.rvRowType.insert(fieldDep, path.toList)._1.asInstanceOf[TStruct]))
         memoizeMatrixIR(child, dep, memo)
+      case MatrixExplodeCols(child, path) =>
+        def getExplodedField(typ: MatrixType): Type = typ.colType.queryTyped(path.toList)._1
+        val baseType = getExplodedField(child.typ)
```

Comment:
`baseType` evokes `BaseType` for me. This is really `preExplosionFieldType`, right?